envi/archs/h8/disasm.py

```diff
@@ -0,0 +1,680 @@
+    Absolute address [@aa:8, @aa:16, or @aa:24]
+    Immediate [#xx:8, #xx:16, or #xx:32]
+    Program-counter relative [@(d:8,PC) or @(d:16,PC)]
+    Memory indirect [@@aa:8]
+
+'''
+def addrToName(mcanv, va):
+    sym = mcanv.syms.getSymByAddr(va)
+    if sym != None:
+        return repr(sym)
+    return "0x%x" % va
+
+class H8Opcode(envi.Opcode):
+    _def_arch = envi.ARCH_H8
+    def __hash__(self):
+        return int(hash(self.mnem) ^ (self.size << 4))
+
+    def __len__(self):
+        return int(self.size)
+
+    def getBranches(self, emu=None):
+        """
+        Return a list of tuples.  Each tuple contains the target VA of the
+        branch, and a possible set of flags showing what type of branch it is.
+
+        See the BR_FOO types for all the supported envi branch flags....
+        Example: for bva,bflags in op.getBranches():
+        """
+        ret = []
+
+        # FIXME: which do we use?  _def_arch?  or  & ARCH_MASK?
+        brflags = (self.iflags & envi.ARCH_MASK) | self._def_arch
+
+        # If we can fall through, reflect that...
+        if not self.iflags & envi.IF_NOFALL:
+            ret.append((self.va + self.size, brflags|envi.BR_FALL))
+
+        # In H8, if we have no operands, it has no
+        # further branches...
+        if len(self.opers) == 0:
+            return ret
+
+        if self.iflags & envi.IF_COND:
+            brflags |= envi.BR_COND
+
+        if self.iflags & envi.IF_BRANCH:
+            if self.opers[0].isDeref():
+                brflags |= envi.BR_DEREF
+            ret.append((self.getOperValue(0), brflags))
+
+        elif self.iflags & envi.IF_CALL:
+            brflags |= envi.BR_PROC
+            if self.opers[0].isDeref():
+                brflags |= envi.BR_DEREF
+            ret.append((self.getOperValue(0), brflags))
+
+        return ret
+
+    def render(self, mcanv):
+        """
+        Render this opcode to the specified memory canvas
+        """
+        mnem = self.mnem
+        if self.iflags & IF_B:
+            mnem += '.b'
+        elif self.iflags & IF_W:
+            mnem += '.w'
+        elif self.iflags & IF_L:
+            mnem += '.l'
+
+        mcanv.addNameText(mnem, typename="mnemonic")
+        mcanv.addText(" ")
+
+        # Allow each of our operands to render
+        imax = len(self.opers)
+        lasti = imax - 1
+        for i in xrange(imax):
+            oper = self.opers[i]
+            oper.render(mcanv, self, i)
+            if i != lasti:
+                mcanv.addText(",")
+
+    def __repr__(self):
+        mnem = self.mnem
+        if self.iflags & IF_B:
+            mnem += '.b'
+        elif self.iflags & IF_W:
+            mnem += '.w'
+        elif self.iflags & IF_L:
+            mnem += '.l'
+        
+        x = []
+        for o in self.opers:
+            x.append(o.repr(self))
+
+        return mnem + " " + ", ".join(x)
+
+
+
+def unittest_parsers(buf = 'ABCDEFGHIJKLMNOP', off=3, va=0x2544):
```

Comment:
If these really are about unit testing, they'll need to be moved ( and implemented as a unittest.TestCase sub class )